Origin: http://selenic.com/repo/hg-stable/rev/035434b407be
Description: test-casefolding.t: demonstrate a bug with HFS+ ignoring some codepoints
 This is a fix for CVE-2014-9390
Applied-Upstream: 3.2.3

--- a/tests/test-casefolding.t
+++ b/tests/test-casefolding.t
@@ -160,4 +160,17 @@ case changes.
   $ hg status -A
   C MiXeDcAsE
 
+#if osx
+
+We assume anyone running the tests on a case-insensitive volume on OS
+X will be using HFS+. If that's not true, this test will fail.
+
+Bug: some codepoints are to be ignored on HFS+:
+
+  $ rm A
+  >>> open(u'a\u200c'.encode('utf-8'), 'w').write('unicode is fun')
+  $ hg status
+  M A
+  ? a\xe2\x80\x8c (esc)
+#endif
   $ cd ..
